public/javascripts/signup/signup.js

```diff
@@ -0,0 +1,78 @@
+(function(){
+	//initialize the angular app and inject dependencies.
+	angular.module("olinbaja.signup", ['ngRoute'])
+    .config(function($routeProvider, $locationProvider) {
+      $routeProvider
+        .when('/signup', {
+          templateUrl: './pages/signup.html',
+          controller: 'SignupController',
+          controllerAs: 'vm'
+        })
+
+      $locationProvider.html5Mode({
+        enabled: true,
+        requireBase: false
+      });
+    })
+    .controller('SignupController', SignupController);
+
+  function SignupController($http, $window) {
+    var vm = this;
```

Comment:
Why are you making a separate reference to this instead of just calling this.foo in the code?